libminifi/include/controllers/keyvalue/AbstractCoreComponentStateManagerProvider.h

```diff
@@ -34,29 +34,39 @@ namespace controllers {
 class AbstractCoreComponentStateManagerProvider : public std::enable_shared_from_this<AbstractCoreComponentStateManagerProvider>,
                                                    public core::CoreComponentStateManagerProvider {
  public:
-  ~AbstractCoreComponentStateManagerProvider() override;
-
   std::shared_ptr<core::CoreComponentStateManager> getCoreComponentStateManager(const utils::Identifier& uuid) override;
-
   std::map<utils::Identifier, std::unordered_map<std::string, std::string>> getAllCoreComponentStates() override;
 
   class AbstractCoreComponentStateManager : public core::CoreComponentStateManager{
    public:
     AbstractCoreComponentStateManager(std::shared_ptr<AbstractCoreComponentStateManagerProvider> provider, const utils::Identifier& id);
 
-    bool set(const core::CoreComponentState& kvs) override;
+    ~AbstractCoreComponentStateManager() override;
 
+    bool set(const core::CoreComponentState& kvs) override;
     bool get(core::CoreComponentState& kvs) override;
-
     bool clear() override;
-
     bool persist() override;
 
+    bool isTransactionInProgress() const override;
+    bool beginTransaction() override;
+    bool commit() override;
+    bool rollback() override;
+
    private:
+    enum class ChangeType {
+      NONE,
+      SET,
+      CLEAR
+    };
+
     std::shared_ptr<AbstractCoreComponentStateManagerProvider> provider_;
     utils::Identifier id_;
     bool state_valid_;
     core::CoreComponentState state_;
+    bool transactionInProgress_;
+    ChangeType changeType_;
+    core::CoreComponentState stateToSet_;
```

Comment:
Use snake_case_ format for member variables

Reply:
Done.